integrations/dd-sdk-android-okhttp/src/test/kotlin/com/datadog/android/okhttp/trace/TracingInterceptorContextInjectionSampledTest.kt

```diff
@@ -0,0 +1,1437 @@
+    lateinit var fakeResponseBody: String
+
+    lateinit var fakeUrl: String
+
+    lateinit var fakeBaseUrl: String
+
+    lateinit var fakeRequest: Request
+    lateinit var fakeResponse: Response
+
+    @StringForgery(type = StringForgeryType.HEXADECIMAL)
+    lateinit var fakeSpanId: String
+
+    @StringForgery(regex = "[a-f][0-9]{32}")
+    lateinit var fakeTraceIdAsString: String
+
+    lateinit var fakeTraceId: BigInteger
+
+    lateinit var fakeOrigin: String
+
+    lateinit var fakeLocalHosts: Map<String, Set<TracingHeaderType>>
+
+    // endregion
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeOrigin = forge.anAlphabeticalString()
+        fakeTraceId = BigInteger(fakeTraceIdAsString, 16)
+        whenever(mockTracer.buildSpan(TracingInterceptor.SPAN_NAME)) doReturn mockSpanBuilder
+        whenever(mockLocalTracer.buildSpan(TracingInterceptor.SPAN_NAME)) doReturn mockSpanBuilder
+        whenever(mockSpanBuilder.withOrigin(fakeOrigin)) doReturn mockSpanBuilder
+        whenever(mockSpanBuilder.asChildOf(null as SpanContext?)) doReturn mockSpanBuilder
+        whenever(mockSpanBuilder.start()) doReturn mockSpan
+        whenever(mockSpan.context()) doReturn mockSpanContext
+        whenever(mockSpanContext.toSpanId()) doReturn fakeSpanId
+        whenever(mockSpanContext.traceId).thenReturn(fakeTraceId)
+        whenever(mockSpanContext.toTraceId()) doReturn fakeTraceId.toString()
+        whenever(mockTraceSampler.sample()) doReturn true
+        val mediaType = forge.anElementFrom("application", "image", "text", "model") +
+            "/" + forge.anAlphabeticalString()
+        fakeLocalHosts =
+            forge.aMap { forge.aStringMatching(HOSTNAME_PATTERN) to setOf(TracingHeaderType.DATADOG) }
+        fakeMediaType = mediaType.toMediaTypeOrNull()
+        fakeUrl = forgeUrlWithQueryParams(forge)
+        fakeRequest = forgeRequest(forge)
+        whenever(rumMonitor.mockSdkCore.getFeature(Feature.TRACING_FEATURE_NAME)) doReturn mock()
+        whenever(rumMonitor.mockSdkCore.internalLogger) doReturn mockInternalLogger
+        whenever(rumMonitor.mockSdkCore.firstPartyHostResolver) doReturn mockResolver
+        testedInterceptor = instantiateTestedInterceptor(fakeLocalHosts) { _, _ ->
+            mockLocalTracer
+        }
+
+        GlobalTracer.registerIfAbsent(mockTracer)
+    }
+
+    fun instantiateTestedInterceptor(
+        tracedHosts: Map<String, Set<TracingHeaderType>> = emptyMap(),
+        factory: (SdkCore, Set<TracingHeaderType>) -> Tracer
+    ): TracingInterceptor {
+        return TracingInterceptor.Builder(tracedHosts)
+            .setTracedRequestListener(mockRequestListener)
+            .setTraceOrigin(fakeOrigin)
+            .setTraceSampler(mockTraceSampler)
+            .setTraceContextInjection(TraceContextInjection.Sampled)
+            .setLocalTracerFactory(factory)
+            .build()
+    }
+
+    fun getExpectedOrigin(): String? {
+        return fakeOrigin
```

Comment:
Minor: Why do we need this method and not compare directly to fakeOrigin? Also, can this return a null value? fakeOrigin is defined as an alphabetical string so should be non-null

Reply:
just legacy code that I did not change